backend-lambda/services/productService.js

```diff
@@ -19,25 +19,54 @@ const getAllProducts = async (pool, limit, offset) => {
 
 // NOTE: This function is untested against cockroach
 const updateProduct = async (pool, id, product) => {
-  let sets = [];
-  for (let key in product) {
-    sets.push(Format("%I = %L", key, product[key]));
-  }
+  poolCheck(pool);
+  productCheck(product);
+
+  const values = Object.values(product);
+  const keys = Object.keys(product);
 
-  let setStrings = sets.join(",");
+  // Creates keys and values array then maps them to one long-ass string ->
+  // ex: 'name = 'purple jumpsuit', size = 'Medium', description = 'comes with pockets', sold_date = '6.29.2024''
+  const setString = keys
+    .map((key, index) => Format("%I = %L", key, values[index]))
+    .join(", ");
 
-  const query = "UPDATE product SET %s WHERE product_id = %L";
-  const params = [setStrings, id];
+  /* 
+  The setString does not require a placeholder in the query nor to be in the parameters variable.
+  This is because the pg library formats the query so that the long-ass string is put into it before
+  it is passed to clientService ->
+  
+  ex:
+  UPDATE product SET name = 'purple jumpsuit', size = 'Medium', description = 'comes with pockets', sold_date = '6.29.2024'
+  WHERE product_id = $1
 
-  return clientService(pool, query, params);
+  The product_id value is the only placeholder used here with postgres's $1 placeholder syntax
+  */
+  const query = `UPDATE product SET ${setString} WHERE product_id = $1`;
+  const params = [id];
+
+  const res = await clientService(pool, query, params);
+  return res?.rows;
 };
 
 // NOTE: This function is untested against cockroach
 const createProduct = async (pool, product) => {
-  const query = "INSERT INTO product (%I) VALUES (%L)";
-  const params = [Object.keys(product), Object.values(product)];
+  const keys = Object.keys(product).join(", ");
+  const values = Object.values(product);
+  // Creates a set of placeholders that matches the count of values
+  const placeholders = values.map((_, i) => `$${i + 1}`).join(", ");
+
+  poolCheck(pool);
```

Comment:
Do the checks first before determining your other consts.

Reply:
rearranged

Reply:
Put your checks above everything. No use in manipulating data if you're not going to need to do anything (i.e. you failed a validation)

Reply:
done:

![image](https://github.com/user-attachments/assets/aaa359cd-a761-462b-b283-80f93628b4b6)